init.lua

```diff
@@ -1186,10 +1200,38 @@ function PaperWM:moveWindow(window, frame)
     end)
 end
 
+---add or remove focused window from the floating layer and retile the space
+function PaperWM:toggleFloating()
+    local window = Window.focusedWindow()
+    if not window then
+        self.logger.d("focused window not found")
+        return
+    end
+
+    local id = window:id()
+    local space = nil
+    is_floating[id] = not is_floating[id]
+    local persisted = {}
+    for k, v in pairs(is_floating) do
+        if v then table.insert(persisted, k) end
+    end
+    hs.settings.set(IsFloatingKey, persisted)
+
+    if is_floating[id] then
+        space = index_table[id]['space']
+        self:removeWindow(window, true)
+    else
+        self:addWindow(window)
+        space = index_table[id]['space']
+    end
+    self:tileSpace(space)
+end
+
 ---supported window movement actions
 PaperWM.actions = {
     stop_events = partial(PaperWM.stop, PaperWM),
     refresh_windows = partial(PaperWM.refreshWindows, PaperWM),
+    toggle_floating = partial(PaperWM.toggleFloating, PaperWM),
```

Comment:
Do you have a suggestion for a default keyboard shortcut for this action? What does Gnome PaperWM use?

Reply:
They use <kbd>Ctrl</kbd><kbd>Super</kbd><kbd>Escape</kbd>; maybe <kbd>⌘</kbd><kbd>⌥</kbd><kbd>⇧</kbd><kbd>Escape</kbd> to better align with the existing defaults? Shift is needed to avoid the os-level binding for <kbd>⌘</kbd><kbd>⌥</kbd><kbd>Escape</kbd> (force quit). Added.